config/proposer/loader.go

```diff
@@ -0,0 +1,288 @@
+package proposer
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/cmd/validator/flags"
+	"github.com/prysmaticlabs/prysm/v5/config"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	validatorService "github.com/prysmaticlabs/prysm/v5/config/validator/service"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/validator"
+	validatorpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1/validator-client"
+	"github.com/prysmaticlabs/prysm/v5/validator/db/iface"
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+type SettingsType int
+
+const (
+	none SettingsType = iota
+	defaultFlag
+	fileFlag
+	urlFlag
+	onlyDB
+)
+
+// ProposerSettingsLoader is an interface for loading proposer settings
+type ProposerSettingsLoader interface {
+	Load(cliCtx *cli.Context) (*validatorService.ProposerSettings, error)
+}
+
+type proposerSettingsLoader struct {
+	LoadMethods []SettingsType
+	ExistsInDB  bool
+	Db          iface.ValidatorDB
+	options     *flagOptions
+}
+
+type flagOptions struct {
+	builderConfig   *validatorService.BuilderConfig
+	gasLimit        validator.Uint64
+	hasGasLimitFlag bool
+}
```

Comment:
```suggestion
type flagOptions struct {
	builderConfig   *validatorService.BuilderConfig
	gasLimit        validator.Uint64
	hasGasLimit bool
}
```
No need to repeat the word `flag` in my opinion as it's already in the type name